### ISSUE TITLE
Error when aws provider is disabled

### ISSUE THREAD

**Issue by joffreychambrin** [pulumi/pulumi-eks#835]
### What happened?

With pulumi-eks version 1.0.0 in python, we have the following error when the aws provider is disabled 
`Exception: Invoke: Default provider for 'aws' disabled. 'aws:index/getPartition:getPartition' must use an explicit provider.`

### Steps to reproduce

Create the following pulumi program 

```python
from pulumi import ResourceOptions
from pulumi_eks import Cluster
from pulumi_aws import Provider

provider = Provider("aws", region="us-west-2")
Cluster("hi", ResourceOptions(providers=[provider]))
```

Disable the aws default provider in the config
```yaml
config:
  pulumi:disable-default-providers:
    - aws
```

### Expected Behavior

No error

### Actual Behavior

`Exception: Invoke: Default provider for 'aws' disabled. 'aws:index/getPartition:getPartition' must use an explicit provider.`

### Output of `pulumi about`

❯ pulumi about
CLI
Version      3.49.0
Go Version   go1.19.4
Go Compiler  gc

Host
OS       darwin
Version  13.0.1
Arch     arm64

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by squaremo** [comment]
Thanks for posting this @joffreychambrin. I think it's reasonable to expect to be able to give an explicit provider, and turn the default provider off. There may be a technical reason why this isn't possible; but on first look, I'm going to leave this as `kind/bug`.

**Comment by bacek** [comment]
Any progress on it? I hit exactly same issue.

**Comment by jaxxstorm** [comment]
I believe this is fixed by #886
